paddle/fluid/imperative/layer.h

```diff
@@ -406,12 +420,54 @@ class OpBase : public std::enable_shared_from_this<OpBase> {
 
  private:
   OpBase(size_t id, const std::string& type, const NameVarBaseMap& ins,
-         const NameVarBaseMap& outs, framework::AttributeMap attrs,
+         const NameVarBaseMap& outs, const framework::AttributeMap& attrs,
          const platform::Place& place);
 
   OpBase(size_t id, const framework::OpDesc& op_desc,
          const platform::Place& place);
 
+ public:
+  OpBase() {}
+
+  void SetType(const std::string& type) { type_ = type; }
+  void SetInput(const std::string& name,
+                std::vector<std::shared_ptr<VarBase>> vec_var_base) {
+    ins_.emplace(name, vec_var_base);
+  }
+  void SetOutput(const std::string& name,
+                 std::vector<std::shared_ptr<VarBase>> vec_var_base) {
+    outs_.emplace(name, vec_var_base);
+  }
+  void SetAttrMap(const framework::AttributeMap& attrs) { attrs_ = attrs; }
+  void SetAttr(const std::string& name, const framework::Attribute& v) {
+    attrs_[name] = v;
+  }
+  void SetBlockAttr(const std::string& name, framework::BlockDesc* block) {}
+
+  const framework::AttributeMap& Attrs() { return attrs_; }
+
+  bool create_operator_base();
+
+  void SetId(size_t id) { id_ = id; }
+  void SetPlace(platform::Place place) { place_ = place; }
+
+  bool HasAttr(const std::string& name) const {
+    return attrs_.find(name) != attrs_.end();
+  }
+
+  framework::Attribute GetAttr(const std::string& name) const {
```

Comment:
Return const reference instead of a copy.

Reply:
done